File: Backend/pokeapp/src/main/java/com/s6x/pokeapp/models/dtos/PokemonDTO.java
package com.s6x.pokeapp.models.dtos;

import com.s6x.pokeapp.models.entities.Ability;
import com.s6x.pokeapp.models.entities.PokemonType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PokemonDTO {
	private Long id;
	private String name;
	private String avatar;
	private PokemonType type;
	private Ability ability;
	private Integer level;
	private Integer health;
	private Integer health_max;
	private CoachDTO coach;
}

File: Frontend/app/src/app/Models/pokemonDTO.ts
export class Pokemon {
    name : string;
    avatar : string;
    type : string;
    ability : string;
    level : number;
    health : number;
    health_max : number;
    coach : number;

    constructor(name : string, avatar : string, type : string, ability : string, level : number, health : number, health_max : number, coach : number){
        this.name = name;
        this.avatar = '../assets/pokemon_images/' + avatar;
        this.type = type;
        this.ability = ability;
        this.level = level || 10;
        this.health = health;
        this.health_max = health || 100;
        this.coach=coach;
    }
}